src/Lecture3.hs

```diff
@@ -79,7 +82,11 @@ Tuesday
   would work for **any** enumeration type in Haskell (e.g. 'Bool',
   'Ordering') and not just 'Weekday'?
 -}
-next = error "TODO"
+next :: Weekday -> Weekday
+next weekday
+  | weekday == maxBound = minBound
+  | otherwise = succ weekday
+-- Sorry, I don't understood how improve my solution for bonus challenge. Maybe you give me advice ? 
```

Comment:
You see now that the body of the function doesn't use any information specific to `Weekday`. The implementation only uses polymorphic methods and values `==`, `maxBound`, `minBound`, `succ`.

So you change the type signature of this function to make it work for more types:

```haskell
next :: ?? constraints here ?? => a -> a
```

Reply:
Maybe like this ?
next :: Enum => a -> a

Reply:
Almost) More like: `next :: (Eq a, Enum a, Bounded a) => a -> a`